cli/lsp/trace.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+use std::fmt;
+
+use crate::lsp::logging::lsp_debug;
+use deno_core::anyhow;
+use opentelemetry::KeyValue;
+use opentelemetry_otlp::WithExportConfig;
+use opentelemetry_sdk::Resource;
+use opentelemetry_semantic_conventions::resource::SERVICE_NAME;
+use serde::Deserialize;
+use serde::Serialize;
+use tracing::level_filters::LevelFilter;
+use tracing_opentelemetry::OpenTelemetryLayer;
+use tracing_subscriber::layer::SubscriberExt;
+use tracing_subscriber::util::SubscriberInitExt;
+
+pub(crate) fn make_tracer(
+) -> Result<opentelemetry_sdk::trace::Tracer, anyhow::Error> {
+  Ok(
+    opentelemetry_otlp::new_pipeline()
+      .tracing()
+      .with_exporter(
+        opentelemetry_otlp::new_exporter()
+          .tonic()
+          .with_endpoint("http://localhost:4317"),
```

Comment:
We probably shouldn't make it a fixed address?

Reply:
Yeah I'll add it to the config